r/R/expression.R

```diff
@@ -25,6 +26,15 @@ runExpression <- function(req, data) {
     # worried about duplicate gene row.names in @data
     symbol_idx <- match(colnames(geneExpression), genesSubset$input)
     colnames(geneExpression) <- genesSubset$name[symbol_idx]
-    #geneExpression <- as.list(as.data.frame(t(as.matrix(geneExpression))))
-    return(geneExpression)
+    adjGeneExpression <- as.data.frame(apply(geneExpression,2,FUN=function(x){
+        lim <- as.numeric(quantile(x,quantile_threshold,na.rm=TRUE))
+        i <- 0.01
+        while (lim==0 & i+quantile_threshold<=1) {
+            lim <- as.numeric(quantile(x,quantile_threshold + i,na.rm=TRUE))
+            i<-i+0.01
+        }
+        print(i)
+        return(pmin(x,lim))
```

Comment:
```suggestion
        return(min(x, lim))
```
`lim` is length 1 so I don't think this needs to be `pmin`

Reply:
We need to compare with the whole x array, so it should be pmin, min doesn't work

Reply:
Ah yes I see now